### DISCUSSION TITLE
File types for mini's various floating windows?

### DISCUSSION THREAD

**Discussion by pkazmier** [nvim-mini/mini.nvim#821] [Q&A]
Would it be possible to add file types for mini's floating windows such as `mini.completion`, `mini.clue`, etc ...? This would allow one to define auto commands for those floating windows based on file type.

The specific reason I'm asking is that I use [reticle](https://github.com/tummetott/reticle.nvim) to ensure the cursor line is only shown in the current window. With that plug-in, you provide a list of file types to ignore. Because the `mini.completion` preview buffer does not have a `filetype`, I am unable to tell reticle to ignore that buffer. As a result, I get a cursor line in my preview window (not mini's fault).

I'm not familiar with standards of neovim plug-in development, so I don't know if adding a `filetype` to mini's floating windows is a good/bad idea. I did notice that `nvim-cmp` floating windows have file types associated with them, so it seems there is some precedent of doing so.

Thoughts?

**Comment by echasnovski** [comment]
In general having custom filetypes for plugin-specific buffers (nit pick for future: filetype is a buffer option, not a window one) is more or less a good idea. However, in some cases it should be done carefully as mere act of setting 'filetype' is quite costly on performance due to many built-in `FileType` autocommands. This is the case in 'mini.completion' and partially in 'mini.clue'.

What I don't really understand is that why that plugin results into those windows having cursor line. It is advertised as working only "in the currently active window" while those windows are never intended to be current. There is a chance they become current for some actions (which I am not sure they do), but eventually and essentially they do not feel for user as current.

**Reply by pkazmier** [reply]
Thanks for the insight. I’ve disabled that plug-in for now. Maybe you can make a mini version of it 😀
